simulation/halsim_xrp/src/main/native/cpp/XRP.cpp

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "XRP.h"
+
+#include <fmt/format.h>
+#include <wpi/Endian.h>
+#include <wpi/MathExtras.h>
+
+using namespace wpilibxrp;
+
+void FloatToBuf(float value, uint8_t* buf) {
+  if constexpr (wpi::support::endian::system_endianness() ==
+                wpi::support::big) {
```

Comment:
It's not actually necessary to do this optimization--the write32be does it internally.  You might consider making this whole function static inline?

Reply:
An ok. If that's the case I could probably just use write32be directly instead of this wrapper?

Reply:
Yes, that would be fine.